arch/sim/src/sim/up_head.c

```diff
@@ -52,6 +53,43 @@ char **g_argv;
 static char g_logbuffer[4096];
 #endif
 
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: allsyms_relocate
+ *
+ * Description:
+ *   Simple relocate to redirect the address from symbol table.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ALLSYMS
+static void allsyms_relocate(void)
+{
+  extern struct symtab_s g_allsyms[1];
+  extern int             g_nallsyms;
```

Comment:
can we move externs out of `allsyms_relocate`?
Also `extern struct symtab_s g_allsyms[];` should work better than `[1]` from warnings point of view perspective.

Reply:
Done